apps/ergw_core/src/ergw_node_selection.erl

```diff
@@ -87,6 +87,9 @@ candidates(Name, Services, NodeSelection) ->
 	    L
     end.
 
+lowercase_apn([H|_] = APN) when is_binary(H) ->
+    [string:lowercase(X) || X <- APN].
```

Comment:
Maybe make sense try use `cowboy_bstr:to_lower/1`

Reply:
probably not the best spot to introduce a dependency on cowboy

Reply:
No,
a) cowboy is not dependency of ergw_core,
b) what does `cowboy_bstr:to_lower/1` better than `string:lowercase/1`

Reply:
```erlang
string:lowercase(<<"привет">>).
<<"?@825b">>
```
bad converting if not utf8.

Reply:
those characters are not permitted in APN names anyway.

yes, we could validate the input a bit further, but failing on this will only make the handler for the current request fail. Since the input is invalid, that is ok.

Reply:
if you truly insisted on using that string as APN, it would need to be encoded as `xn--b1agh1afp` to be valid. UI might then choose to actually display that as unicode string.

Reply:
I see, thanks.